Cleans up/renames as private some internal var definitions, removes some fucked uses of internal list vars

LemonInTheDark opened the pull request:
## About The Pull Request

[Improves the documentation of DCS lists, removes old list of callback docs that no longer apply](https://github.com/tgstation/tgstation/commit/c3821d9f5ffaeaa4772f927c819da0c1de0ca27c)

[Adds a second signal register to decal rotating, adds a trait to objects under a tile. STOP DIRECTLY READING HIDDEN LISTS I SWEAR TO GOD](https://github.com/tgstation/tgstation/commit/6b3f97a76a6f7d24ab952739a1561633922994e1)

[Removes direct reads of the timer list, they were redundant mostly](https://github.com/tgstation/tgstation/commit/14fcd9f8a6d1b2d42ec6df3493ebc76fe7c12032)

[Please stop directly reading/modifying the traits list to ensure your dna rot follows the brain](https://github.com/tgstation/tgstation/commit/ec0e5237ec2b7c3b7806cb993670acc8ce388bdc)

[Marks internal datum lists as well internal with _](https://github.com/tgstation/tgstation/pull/75769/commits/57c6577ff61629b8ea792ee37ec4f2490a8e2865)
[57c6577](https://github.com/tgstation/tgstation/pull/75769/commits/57c6577ff61629b8ea792ee37ec4f2490a8e2865)

Does the same to _clear_signal_refs() in hopes of keeping people from touching it

## Why It's Good For The Game

They pissed me off.

Users should not be touching these lists, especially in ways that make assumptions about their structure and are thus prone to breaking if that ever changes.
Most of these are close to zero cost changes, using a wrapper to solve the problem, or just yeeting it

Two aren't, Decals with a direction have gained a second signal register on init, and things that sit underfloor (cables/pipes) now get a trait when inserted there.

This should have a minimal impact on memory/init time, bugging @Mothblocks about it just in case

LemonInTheDark committed:
Improves the documentation of DCS lists, removes old list of callback docs that no longer apply

LemonInTheDark committed:
Adds a second signal register to decal rotating, adds a trait to objects under a tile. STOP DIRECTLY READING HIDDEN LISTS I SWEAR TO GOD

LemonInTheDark committed:
Removes direct reads of the timer list, they were redundant mostly

LemonInTheDark committed:
Please stop directly reading/modifying the traits list to ensure your dna rot follows the brain

LemonInTheDark committed:
Whoops

LemonInTheDark committed:
oh come on man

MrStonedOne commented:
> [Removes direct reads of the timer list, they were redundant mostly](https://github.com/tgstation/tgstation/commit/14fcd9f8a6d1b2d42ec6df3493ebc76fe7c12032)

> [Please stop directly reading/modifying the traits list to ensure your dna rot follows the brain](https://github.com/tgstation/tgstation/commit/ec0e5237ec2b7c3b7806cb993670acc8ce388bdc)

protip: No var that doesn't start with _ is untouchable. 

there is a easy solution to that.

LemonInTheDark committed:
Marks internal datum lists as well internal with _

Does the same to _clear_signal_refs() in hopes of keeping people from
touching it

LemonInTheDark commented:
Good idea
